ceph_iscsi_config/client.py

```diff
@@ -256,7 +268,11 @@ def _add_lun(self, image, lun):
         if image in self.metadata['luns'].keys():
             lun_id = self.metadata['luns'][image]['lun_id']
         else:
-            lun_id = self.lun_id_list[0]     # pick the lowest available lun ID
+            if image in self.lun_lookup:
+                # this indicates a lun map for a group managed client
+                lun_id = self.lun_lookup[image]
+            else:
+                lun_id = self.lun_id_list[0]  # pick lowest available lun ID
```

Comment:
Semi-related: does a lun ever get re-added to the ```lun_id_list``` if the lun is deleted?

Reply:
The creation of a client instance is transitory - which is part of the design for ansible based deployments. A client object is created, acted upon to bring the config in line and then the instance is just garbage collected. The lun_id_list is first created with all available lun id's and then the lun id's assigned to the client (from the gateway.conf) are removed from the available range. A request for a new lun for the client, then start at the first available lun id.

Reply:
Ok -- just wanted to make sure the API wasn't caching it in-memory